### ISSUE TITLE
Push to a preexisting branch when calling `fern generate` with `mode: pull-request` 

### ISSUE THREAD

**Issue by dvargas92495** [fern-api/fern#2376]
**Is your feature request related to a problem? Please describe.**
I'm iterating on a new API and would like to test how it looks on the client. I don't want to generate straight to staging as that could block others on the team or be overridden. I have been using `mode:pull-request` to have my own isolated branch to test these changes. However, as I iterate, I'd like to update my existing branch as opposed to creating new pull requests on each `fern generate` call.

**Describe the solution you'd like**
A way to specify the name of the branch to generate to.

**Describe alternatives you've considered**
An alternative could be to generate locally, then push to a branch after the fact.

**Additional context**
See `fern-x-vellum` channel for more context

**Comment by dannysheridan** [comment]
@dvargas92495 did you end up finding a workaround that unblocked you here?

**Comment by dvargas92495** [comment]
yea this won't block us, think it's a neat idea still though. Super low pri

**Comment by dannysheridan** [comment]
closing as this is low priority and we haven't gotten the request again. neat idea for a clean DX though!

**Comment by dsinghvi** [comment]
@dvargas92495 this should be supported in the existing `generators.yml` https://github.com/fern-api/fern/blob/0af503e7134fcd6a505fd90101d376b7b21b6e8d/fern/apis/generators-yml/definition/group.yml#L63-L68

```
github: 
  mode: push
  branch: my-branch
  repository: my-org/my-branch
```
